supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -59,6 +62,110 @@ case object Cancelled extends TesRunStatus {
+       |BEARER_TOKEN="$${BEARER_TOKEN%\\"}"
+       |
+       |# Use the precomputed endpoint from cromwell + WSM to acquire a sas token
+       |echo Requesting sas token from WSM...
+       |sas_response_json=$$(curl -s \\
+       |                    --retry 3 \\
+       |                    --retry-delay 2 \\
+       |                    -X POST "$getSasWsmEndpoint" \\
+       |                    -H "Content-Type: application/json" \\
+       |                    -H "accept: */*" \\
+       |                    -H "Authorization: Bearer $${BEARER_TOKEN}")
+       |
+       |# Store token as environment variable
+       |export $environmentVariableName=$$(echo "$${sas_response_json}" | jq -r '.token')
+       |
+       |# Echo the first characters for logging/debugging purposes. "null" indicates something went wrong.
+       |echo Saving sas token: $${$environmentVariableName:0:4}**** to environment variable $environmentVariableName...
+       |### END ACQUIRE LOCAL SAS TOKEN ###
+       |""".stripMargin
+  }
+
+  private def maybeConvertToBlob(pathToTest: Try[Path]): Try[BlobPath] = {
+    pathToTest.collect { case blob: BlobPath => blob }
+  }
+
+  /**
+   * Computes an endpoint that can be used to retrieve a sas token for a particular blob storage container.
+   * This assumes that some of the task inputs are blob files, all blob files are in the same container, and we can get a sas
+   * token for this container from WSM.
+   * The task VM will use the user assigned managed identity that it is running as in order to authenticate.
+   * @param taskInputs The inputs to this particular TesTask. If any are blob files, the first  will be used to
+   *                   determine the storage container to retrieve the sas token for.
+   * @param pathGetter A function to convert string filepath into a cromwell Path object.
+   * @param blobConverter A function to convert a Path into a Blob path, if possible. Provided for testing purposes.
+   * @return A URL endpoint that, when called with proper authentication, will return a sas token.
+   *         Returns 'None' if one should not be used for this task.
+   */
+  def determineWSMSasEndpointFromInputs(taskInputs: List[Input],
+                                        pathGetter: String => Try[Path],
+                                        logger: JobLogger,
+                                        blobConverter: Try[Path] => Try[BlobPath] = maybeConvertToBlob): Try[String] = {
+    // Collect all of the inputs that are valid blob paths
+    val blobFiles = taskInputs
+      .collect{ case Input(_, _, Some(url), _, _, _) => blobConverter(pathGetter(url)) }
+      .collect{ case Success(blob) => blob }
+
+    // Log if not all input files live in the same container.
+    if (blobFiles.map(_.container).distinct.size > 1) {
+      logger.info(s"While parsing blob inputs, found more than one container. Can only generate an environment sas token for a single blob container at once.")
```

Comment:
The original wording of this error doesn't make clear that we'll actually try making a token, as opposed to detecting this condition and erroring out.
```suggestion
      logger.info(s"While parsing blob inputs, found more than one container. Generating SAS token based on first file in the list.")
```